### PR TITLE
Stef -- Updated Grocery Store -- Carets

### PR DESCRIPTION
**Pull request by SesameSeeds**
# Grocery Store
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Question 	|   Response	|
|---	|---	|
|   Why is it useful to put classes inside modules? 	|   	|
|   What is accomplished with `raise ArgumentError`? 	|   	|
|   Why do you think we made the `.all` & `.find` methods class methods?  Why not instance methods?	|   |
|   Why does it make sense to use inheritance for the online order?  |  |
|   Did the presence of automated tests change the way you thought about the problem? How? | |

### CODE REVIEW

**Comment by CheezItMan** [comment]
1st note, add `coverage` to a `.gitignore` file, that way the coverage data won't be in the pull request.

File: specs/order_spec.rb [hunk 1]
```diff
@@ -78,32 +93,50 @@
...
+    end
+
+    it "Returns the first order from the CSV file" do
+      order = @orders.first
+      order.id.must_equal 1
+    end
+
+    it "Returns the last order from the CSV file" do
+      order = @orders.last
+      order.id.must_equal 100
```
Comment (CheezItMan) [inline specs/order_spec.rb:122]: Good, you should also check the products in the order to verify that they're right.

File: specs/order_spec.rb [hunk 2]
```diff
@@ -78,32 +93,50 @@
...
 
     it "Can find the last order from the CSV" do
-      # TODO: Your test code here!
+      Grocery::Order.all
+      Grocery::Order.find(100).id.must_equal 100
     end
 
     it "Raises an error for an order that doesn't exist" do
-      # TODO: Your test code here!
+      proc { Grocery::Order.find(1000) }.must_raise ArgumentError
```
Comment (CheezItMan) [inline specs/order_spec.rb:138]: I would also test with a negative order and one greater than the last element.

File: lib/order.rb [hunk 3]
```diff
@@ -7,12 +9,58 @@ def initialize(id, products)
...
+        pairs = products.split(";")
+
+        products_hash = {}
+        pairs.each do |pair|
+          item = pair.split(":")
+          products_hash[item[0]] = item[1].to_f
+        end
+
+        orders << self.new(id, products_hash)
+      end
```
Comment (CheezItMan) [inline lib/order.rb:29]: Very nicely done, I like the `self.new`.

**Review by CheezItMan — COMMENTED** [review]
Really good work, well done on both the spec and the `.all` and `.find` methods.
